atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -123,6 +124,27 @@ public static SerializableTransactionManager create(
             Environment env,
             LockServerOptions lockServerOptions,
             boolean allowHiddenTableAccess) {
+        return create(config, schemas, env, lockServerOptions, allowHiddenTableAccess, UserAgents.DEFAULT_USER_AGENT);
+    }
+
+    public static SerializableTransactionManager create(
+            AtlasDbConfig config,
+            Set<Schema> schemas,
+            Environment env,
+            LockServerOptions lockServerOptions,
+            boolean allowHiddenTableAccess,
+            Class<?> callingClass) {
+        return create(config, schemas, env, lockServerOptions, allowHiddenTableAccess,
+                UserAgents.fromClass(callingClass));
+    }
+
+    public static SerializableTransactionManager create(
+            AtlasDbConfig config,
+            Set<Schema> schemas,
+            Environment env,
+            LockServerOptions lockServerOptions,
+            boolean allowHiddenTableAccess,
+            String userAgent) {
```

Comment:
I'm in two minds about whether the `create` and `createLockAndTimestamp` services that take a `String userAgent` need to be in the public API. 

In the spirit of keeping the API small, could we make them private? (I just tried recompiling with this as private but my IntelliJ is borked :(.)